lib/presentation/widget/enter_otp/enter_otp_widget.dart

```diff
@@ -48,45 +58,67 @@ class EnterOTPWidget extends StatelessWidget {
-                icon: Image(
-                    image: AssetImage(imagePath.icArrowBack),
-                    alignment: Alignment.center),
-              ),
-            ),
-            Align(
-              alignment: Alignment.center,
-              child: SizedBox(
-                height: 200,
-                child: Column(
-                  children: [
-                    SizedBox(
-                      child: TextField(
-                        controller: textController,
-                      ),
-                      width: 200,
-                    ),
-                    GestureDetector(
-                      onTap: () => enterOTPViewModel.onLoginPressed(textController.text),
-                      child: Padding(
-                        padding: const EdgeInsets.all(20.0),
-                        child: Text('submit'),
-                      ),
-                    )
-                  ],
+      body: GestureDetector(
+        onTap: () => FocusScope.of(context).unfocus(),
+        child: SafeArea(
+          child: Stack(children: [
+            SingleChildScrollView(
+              reverse: false,
+              child: Padding(
+                padding: EdgeInsets.only(bottom: MediaQuery.of(context).viewInsets.bottom),
+                child: Container(
+                  alignment: Alignment.topCenter,
+                  padding: EdgeInsets.only(top: 100),
+                  child: Column(
+                    children: [
+                      LinShareSloganBuilder()
+                          .setSloganText(AppLocalizations.of(context).login_text_slogan)
+                          .setSloganTextAlign(TextAlign.center)
+                          .setSloganTextStyle(TextStyle(color: Colors.white, fontSize: 16))
+                          .setLogo(imagePath.icLoginLogo)
+                          .build(),
+                      SizedBox(height: 80),
+                      PinCodeWidget.twoFactorAuthen(
+                          title: AppLocalizations.of(context).second_factor_authentication,
+                          titleStyle: _titleTextStyle,
+                          subTitle: AppLocalizations.of(context).input_6_digit_from_free_otp,
+                          subTitleStyle: _subTitleTextStyle,
+                          pinCodeFieldBackgroundColor: AppColor.primaryColor,
+                          pinBackgroundColor: AppColor.pinCodePadBackgroundColor,
+                          pinTextStyle: _padTextStyle,
+                          errorText: AppLocalizations.of(context).please_fill_up_all_numbers,
+                          errorStyle: _errorTextStyle,
+                          buttonText: AppLocalizations.of(context).submit,
+                          buttonBackgroundColor: AppColor.pinCodeSubmitButtonBGColor,
+                          buttonStyle: _buttonActionTextStyle,
+                          onButtonClick: (enteredPin) => _onSubmitPressed(enteredPin, context),
```

Comment:
I think you should call `enterOTPViewModel.onLoginPressed(otpCode, context)` directly from here

Reply:
I want to separate them from the widget tree. This makes the widget tree to be cleaner. IMO